### PR TITLE
tesseract5: fix darwin compilation

### PR DESCRIPTION
**Pull request by viraptor**
###### Description of changes

Update tesseract5 to include the required darwin frameworks.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/tesseract/tesseract5.nix [hunk 1]
```diff
@@ -1,5 +1,7 @@
 { lib, stdenv, fetchFromGitHub, autoreconfHook, autoconf-archive, pkg-config
-, leptonica, libpng, libtiff, icu, pango, opencl-headers, fetchpatch }:
+, leptonica, libpng, libtiff, icu, pango, opencl-headers, fetchpatch
+, darwin
```
Comment (viraptor) [inline pkgs/applications/graphics/tesseract/tesseract5.nix:3]: This is not how we normally do things, however in this case it's a solution that doesn't require complicated rewrite of https://github.com/NixOS/nixpkgs/blob/07d567ed0ae7981fab5b0dae27da9ebca9fd7ed5/pkgs/applications/graphics/tesseract/wrapper.nix

Passing the frameworks separately the way we do it in all-packages requires either:
- the wrapper to ignore them for tesseract3/4,
- tesseract3/4 to accept them and leave unused, or
- using `...` to ignore them

Neither of those seem better than making an exception for accepting darwin here.

Feel free to suggest other solutions though.

File: pkgs/applications/graphics/tesseract/tesseract5.nix [hunk 2]
```diff
@@ -27,13 +29,17 @@ stdenv.mkDerivation rec {
...
+    CoreVideo
+  ]);
 
   meta = {
     description = "OCR engine";
     homepage = "https://github.com/tesseract-ocr/tesseract";
     license = lib.licenses.asl20;
     maintainers = with lib.maintainers; [ anselmschueler ];
-    platforms = with lib.platforms; linux ++ darwin;
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```
Comment (viraptor) [inline pkgs/applications/graphics/tesseract/tesseract5.nix:43]: referencing just `darwin` here uses the `pkgs.darwin` instead of the platform.

**Comment by schuelermine** [comment]
Thank you!

**Review by schuelermine — APPROVED** [review]
Can't check as I don't have a Mac, looks fine

File: pkgs/applications/graphics/tesseract/tesseract5.nix [hunk 3]
```diff
@@ -27,13 +29,17 @@ stdenv.mkDerivation rec {
     icu
     pango
     opencl-headers
-  ];
+  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/tesseract/tesseract5.nix:32]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```

File: pkgs/applications/graphics/tesseract/tesseract5.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/tesseract/tesseract5.nix:43]:
```suggestion
    platforms = lib.platforms.unix;
```

File: pkgs/applications/graphics/tesseract/tesseract5.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/tesseract/tesseract5.nix:3]: > it's a solution that doesn't require complicated rewrite



> using `...` to ignore them

Adding two ... is not complicated.

Comment (viraptor) [reply pkgs/applications/graphics/tesseract/tesseract5.nix]: Turns out I made it way harder than it needed to be. No `...`s necessary.

**Comment by viraptor** [comment]
@SuperSandro2000 I think that's a more sane way to do it now. Validated compiling all versions on darwin.

**Pull request merged**